### PR TITLE
Fix checkpoint loading and checkpoint tests

### LINKED ISSUE #170: Use distributed when world_size=1 if requested
Right now, we disable distributed functionality if we are using a single gpu: https://github.com/mlfoundations/open_lm/blob/main/open_lm/distributed.py#L20-L25. But if WORLD_SIZE is provided, we should behave as if we are in a distributed environment, which in turn will allow us to run tests that verify the distributed code paths without requiring multiple gpus.

### PR DESCRIPTION
**Pull request by achalddave**
This addresses a few issues with our checkpoint save/load tests, and with checkpoint loading itself:

1. Adds a new test that calls main() to test checkpoint saving and loading, ensuring that our full main() pipeline works correctly.
2. Adds a `--force-distributed` flag that allows us to test distributed code paths on a single gpu. Previously, all our tests were running with args.distributed=False, even when args.fsdp=True.
3. Reverts a bug introduced by #153, which meant we did not automatically resume the latest checkpoint when using s3.
4. Fixes a bug in `--resume latest` which led to checkpoints being ignored when resuming in the case of local checkpoints without remote-sync.

Closes #170

### CODE REVIEW

File: open_lm/main.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def natural_key(string_):
 def get_latest_checkpoint(path: str):
     is_s3 = path.startswith("s3")
     fs, root_path = fsspec.core.url_to_fs(path)
-    checkpoints = fs.glob(os.path.join(root_path, "**/epoch_*.pt"))
+    checkpoints = fs.glob(os.path.join(root_path, "epoch_*.pt"))
```
Comment (sagadre) [inline open_lm/main.py:84]: is root_path here what is referred to as checkpoint_path elsewhere?

Comment (achalddave) [reply open_lm/main.py:84]: yup!

File: open_lm/main.py [hunk 2]
```diff
@@ -355,6 +362,10 @@ def main(args):
         resume_from = None
         checkpoint_path = args.checkpoint_path
 
+        # If using remote_sync, need to check the remote instead of the local checkpoints folder.
+        if args.remote_sync is not None:
+            checkpoint_path = os.path.join(args.remote_sync, args.name, "checkpoints")
```
Comment (achalddave) [inline open_lm/main.py:367]: One question here @sagadre - we assign checkpoint_path, but we don't update args.checkpoint_path. I copied this from code from before #153. Should we just go ahead and update args.checkpoint_path too? Not sure if we use it anywhere else.

Comment (sagadre) [reply open_lm/main.py:367]: looked over the code and this should work and not break other things. my understanding is checkpoint_path is used just to load the checkpoint, e.g., latest checkpoint from s3; however, in general we probably should simplify the checkpoint loading in a separate PR. right now it is a bit confusing with checkpoint_path, args.checkpoint_path, join(.., checkpoints), remote-sync, etc.

Comment (sagadre) [reply open_lm/main.py:367]: maybe just having one factorized helper function to do all this loading business could be nice and also more easily unit-testable

**Review by sagadre — APPROVED** [review]

**Pull request merged**